src/scripts/blah_load_config.sh

```diff
@@ -45,20 +45,32 @@ elif [ -d "${GLITE_LOCATION:-/opt/glite}/libexec" ]; then
 else
   blah_libexec_directory="/usr/libexec"
 fi
+# If the libexec directory has a 'blahp' subdirectory, then the blahp
+# files should be there.
+if [ -d "${blah_libexec_directory}/blahp" ]; then
+  blah_libexec_directory="${blah_libexec_directory}/blahp"
+fi
 
-# Let blah_bin_directory be overridden in the config file.
+# Let blah_{bin|sbin|libexec}_directory be overridden in the config file.
 
+config_file=/etc/blah.config
 if [ -r "$BLAHPD_CONFIG_LOCATION" ]; then
-  . $BLAHPD_CONFIG_LOCATION
-elif [ -r "${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config" ]; then
-  . ${GLITE_LOCATION:-/opt/glite}/etc/batch_gahp.config
-elif [ -r "${BLAHPD_LOCATION}/etc/batch_gahp.config" ]; then
-#   This will default to /etc/batch_gahp.config if BLAHPD_LOCATION is unset.
-  . ${BLAHPD_LOCATION}/etc/batch_gahp.config
+  config_file=$BLAHPD_CONFIG_LOCATION
+elif [ "x$BLAHPD_LOCATION" != "x" -a -r "${BLAHPD_LOCATION}/etc/blah.config" ]; then
+  config_file=${BLAHPD_LOCATION}/etc/blah.config
 elif [ -r "${GLITE_LOCATION:-/opt/glite}/etc/blah.config" ]; then
-  . ${GLITE_LOCATION:-/opt/glite}/etc/blah.config
-elif [ -r "${BLAHPD_LOCATION}/etc/blah.config" ]; then
-#   This will default to /etc/blah.config if BLAHPD_LOCATION is unset.
-  . ${BLAHPD_LOCATION}/etc/blah.config
+  config_file=${GLITE_LOCATION:-/opt/glite}/etc/blah.config
 fi
+config_dir=${config_file}.d
+
+. $config_file
 
+if [ -d "$config_dir" ] ; then
+  for file in `ls $config_dir` ; do
+    . ${config_dir}/${file}
```

Comment:
Admittedly the blahp does have some interesting features, but unless you intend shell interpretation of filenames with whitespace and/or glob chars to be one of them, i must insist on safer filename handling:

```suggestion
  for file in "${config_dir}"/* ; do
    . "${config_dir}/${file}"
```

(though the curlies around config_dir is just to match the rest of the styling)